_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -286,31 +278,15 @@ mfxStatus mfxSchedulerCore::Synchronize(mfxTaskHandle handle, mfxU32 timeToWait)
     }
     else
     {
-        UMC::AutomaticMutex guard(m_guard);
-        vm_status res;
-        vm_tick start_tick = vm_time_get_tick(), end_tick;
-        mfxU32 spent_ms;
-
-        while ((pTask->jobID == handle.jobID) &&
-               (MFX_WRN_IN_EXECUTION == pTask->opRes)) {
-            MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_PRIVATE, "Scheduler::Wait");
-            MFX_LTRACE_1(MFX_TRACE_LEVEL_SCHED, "^Depends^on", "%d", pTask->param.task.nParentId);
-            MFX_LTRACE_I(MFX_TRACE_LEVEL_SCHED, timeToWait);
-
-            res = vm_cond_timedwait(&pTask->done, &m_guard, timeToWait);
-            if ((VM_OK == res) || (VM_TIMEOUT == res)) {
-                end_tick = vm_time_get_tick();
-
-                spent_ms = (mfxU32)((end_tick - start_tick)/m_vmtick_msec_frequency);
-                if (spent_ms >= timeToWait) {
-                  break;
-                }
-                timeToWait -= spent_ms;
-                start_tick = end_tick;
-            } else {
-                return MFX_ERR_UNKNOWN;
-            }
-        }
+        std::unique_lock<std::mutex> guard(m_guard);
+
+        MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_PRIVATE, "Scheduler::Wait");
+        MFX_LTRACE_1(MFX_TRACE_LEVEL_SCHED, "^Depends^on", "%d", pTask->param.task.nParentId);
+        MFX_LTRACE_I(MFX_TRACE_LEVEL_SCHED, timeToWait);
+
+        pTask->done.wait_for(guard, std::chrono::milliseconds(timeToWait), [&pTask, &handle] {
```

Comment:
@mgonchar : what's your recommendation for this capture?

Reply:
As I see mfxTaskHandle size is sizeof(size_t), so it definitely can be captured by value, as well as pTask